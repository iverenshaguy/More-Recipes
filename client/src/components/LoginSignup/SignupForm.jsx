import React from 'react';
import { TabPane, Form, Label, Input, FormGroup, FormText, Button } from 'reactstrap';

const placeholder = '\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022\u2022';
const SignupForm = () => (
  <TabPane tabId="2" id="register-form-div">
    <h4 className="text-center">Register for a New Account</h4>
    <hr />
    <p className="text-center">
      <small className="text-muted mx-auto text-center">
        Fields marked
        <span className="text-danger">*</span> are important
      </small>
    </p>
    <Form id="register-form" className="mt-4 mb-3 px-5">
      <FormGroup>
        <Label for="firstname" className="col-form-label col-form-label-lg">
          First Name
          <span className="text-danger">*</span>
        </Label>
        <Input type="text" id="firstname" placeholder="Jane" />
      </FormGroup>
      <FormGroup>
        <Label for="lastname" className="col-form-label col-form-label-lg">
          Last Name
        </Label>
        <Input type="text" id="lastname" placeholder="Smith" />
      </FormGroup>
      <FormGroup>
        <Label for="username" className="col-form-label col-form-label-lg">
          Username
          <span className="text-danger">*</span>
        </Label>
        <Input type="text" id="username" placeholder="janesmith" />
      </FormGroup>
      <FormGroup>
        <Label for="email" className="col-form-label col-form-label-lg">
          Email Address
          <span className="text-danger">*</span>
        </Label>
        <Input type="email" id="email" placeholder="janesmith@me.com" />
      </FormGroup>
      <FormGroup>
        <Label for="signup-password" className="col-form-label col-form-label-lg">
          Password
          <span className="text-danger">*</span>
        </Label>
        <Input type="password" id="signup-password" placeholder={placeholder} />
        <FormText tag="small" className="text-muted">
          Password should be more than 8 characters!
        </FormText>
      </FormGroup>
      <FormGroup>
        <Label for="signup-passwordconfirm" className="col-form-label col-form-label-lg">
          Password Confirmation
          <span className="text-danger">*</span>
        </Label>
        <Input type="password" id="signup-passwordconfirm" placeholder={placeholder} />
      </FormGroup>
      <FormGroup>
        <Label htmlFor="user-info" className="col-form-label col-form-label-lg">
          Tell Us Something About Yourself
        </Label>
        <Input type="textarea" rows={3} id="user-info" placeholder="I love to cook..." />
      </FormGroup>
      <FormGroup>
        <Label htmlFor="occupation" className="col-form-label col-form-label-lg">
          Occupation
        </Label>
        <Input type="text" id="occupation" placeholder="Chef" />
      </FormGroup>
      <Button className="btn-block">REGISTER</Button>
    </Form>
  </TabPane>
);

export default SignupForm;
