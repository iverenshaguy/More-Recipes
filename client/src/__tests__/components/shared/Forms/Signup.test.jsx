import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import SignupForm from '../../../../components//shared/Forms/SignupForm';

const dispatchMock = jest.fn();

const setup = () => {
  const props = {
    state: {
      values: {
        firstname: '',
        lastname: '',
        username: '',
        email: '',
        password: '',
        passwordConfirm: '',
        aboutMe: '',
        occupation: ''
      },
      touched: {
        firstname: false,
        lastname: false,
        username: false,
        email: false,
        password: false,
        passwordConfirm: false,
        aboutMe: false,
        occupation: false
      },
      error: {
        firstname: null,
        lastname: null,
        username: null,
        email: null,
        password: null,
        passwordConfirm: null,
        aboutMe: null,
        occupation: null
      },
      pristine: true,
      formValid: false,
      asyncValidating: false,
      submitting: false,
      submitError: null
    },
    handlers: {
      handleBlur: jest.fn(),
      handleChange: jest.fn(),
      handleFocus: jest.fn(),
      handleSubmit: jest.fn()
    }

  };

  const mountRoot = mount( //eslint-disable-line
    <MemoryRouter>
      <SignupForm {...props} dispatch={dispatchMock} />
    </MemoryRouter>);

  return { props, mountRoot };
};

describe('SignupForm', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { props } = setup();
    const shallowComponent = shallow(<SignupForm {...props} />);

    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

  it('disables submit button when form is clean', () => {
    const { props } = setup();
    const shallowComponent = shallow(<SignupForm {...props} />);

    expect(shallowComponent.find('Button[disabled=true]')).toBeTruthy();
  });

  it('disables submit button when submitting form', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      state: { ...props.state, submitting: true }
    };
    const shallowComponent = shallow(<SignupForm {...newProps} />);

    expect(toJson(shallowComponent)).toMatchSnapshot();
    expect(shallowComponent.find('Button[disabled=true]')).toBeTruthy();
  });

  it('disables submit button when form is invalid', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      state: { ...props.state, formValid: false }
    };
    const shallowComponent = shallow(<SignupForm {...newProps} />);

    expect(toJson(shallowComponent)).toMatchSnapshot();
    expect(shallowComponent.find('Button[disabled=true]')).toBeTruthy();
  });

  it('shows error alert and disables submit button when there\'s a submit error', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      state: { ...props.state, submitError: 'Error submitting form' }
    };

    const shallowComponent = shallow(<SignupForm {...newProps} />);

    expect(toJson(shallowComponent)).toMatchSnapshot();
    expect(shallowComponent.find('NormalAlert')).toBeTruthy();
    expect(shallowComponent.find('Button[disabled=true]')).toBeTruthy();
  });
});
