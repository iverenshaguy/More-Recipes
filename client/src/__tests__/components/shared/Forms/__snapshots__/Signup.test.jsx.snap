// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm disables submit button when form is invalid 1`] = `
<div
  id="register-form-div"
>
  <h4
    className="text-center"
  >
    Register for a New Account
  </h4>
  <hr />
  <p
    className="text-muted mx-auto text-center"
  >
    Fields marked
    <span
      className="text-danger"
    >
      *
    </span>
     are important
  </p>
  <Form
    className="mt-4 mb-3 px-5"
    id="register-form"
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="firstname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        First Name
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="firstname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="firstname"
        placeholder="Jane"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="lastname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Last Name
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="lastname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="lastname"
        placeholder="Smith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email Address
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="email"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="email"
        placeholder="janesmith@me.com"
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="username"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Username
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="username"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="username"
        placeholder="janesmith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
       
      <small
        className="text-muted"
      >
        (more than 10 characters)
      </small>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="password"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="password"
        placeholder="••••••••••"
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="passwordconfirm"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password Confirmation
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="passwordconfirm"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="passwordConfirm"
        placeholder={null}
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="user-info"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Tell Us Something About Yourself
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="user-info"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="aboutMe"
        placeholder="I love to cook"
        type="textarea"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="occupation"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Occupation
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="occupation"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="occupation"
        placeholder="Chef"
        type="text"
        value=""
      />
    </FormGroup>
    <Button
      className="btn-block mt-0"
      color="secondary"
      disabled={true}
      tag="button"
    >
      REGISTER
    </Button>
  </Form>
  <p
    className="text-center"
  >
    Already have an account, 
    <Link
      replace={false}
      target="_self"
      to="\\\\login"
    >
      Log in here
    </Link>
    .
  </p>
</div>
`;

exports[`SignupForm disables submit button when submitting form 1`] = `
<div
  id="register-form-div"
>
  <h4
    className="text-center"
  >
    Register for a New Account
  </h4>
  <hr />
  <p
    className="text-muted mx-auto text-center"
  >
    Fields marked
    <span
      className="text-danger"
    >
      *
    </span>
     are important
  </p>
  <Form
    className="mt-4 mb-3 px-5"
    id="register-form"
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="firstname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        First Name
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="firstname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="firstname"
        placeholder="Jane"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="lastname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Last Name
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="lastname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="lastname"
        placeholder="Smith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email Address
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="email"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="email"
        placeholder="janesmith@me.com"
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="username"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Username
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="username"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="username"
        placeholder="janesmith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
       
      <small
        className="text-muted"
      >
        (more than 10 characters)
      </small>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="password"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="password"
        placeholder="••••••••••"
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="passwordconfirm"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password Confirmation
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="passwordconfirm"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="passwordConfirm"
        placeholder={null}
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="user-info"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Tell Us Something About Yourself
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="user-info"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="aboutMe"
        placeholder="I love to cook"
        type="textarea"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="occupation"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Occupation
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="occupation"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="occupation"
        placeholder="Chef"
        type="text"
        value=""
      />
    </FormGroup>
    <Button
      className="btn-block mt-0"
      color="secondary"
      disabled={true}
      tag="button"
    >
      REGISTER
    </Button>
  </Form>
  <p
    className="text-center"
  >
    Already have an account, 
    <Link
      replace={false}
      target="_self"
      to="\\\\login"
    >
      Log in here
    </Link>
    .
  </p>
</div>
`;

exports[`SignupForm renders correctly 1`] = `
<div
  id="register-form-div"
>
  <h4
    className="text-center"
  >
    Register for a New Account
  </h4>
  <hr />
  <p
    className="text-muted mx-auto text-center"
  >
    Fields marked
    <span
      className="text-danger"
    >
      *
    </span>
     are important
  </p>
  <Form
    className="mt-4 mb-3 px-5"
    id="register-form"
    onSubmit={[Function]}
    tag="form"
  >
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="firstname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        First Name
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="firstname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="firstname"
        placeholder="Jane"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="lastname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Last Name
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="lastname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="lastname"
        placeholder="Smith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email Address
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="email"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="email"
        placeholder="janesmith@me.com"
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="username"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Username
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="username"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="username"
        placeholder="janesmith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
       
      <small
        className="text-muted"
      >
        (more than 10 characters)
      </small>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="password"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="password"
        placeholder="••••••••••"
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="passwordconfirm"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password Confirmation
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="passwordconfirm"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="passwordConfirm"
        placeholder={null}
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="user-info"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Tell Us Something About Yourself
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="user-info"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="aboutMe"
        placeholder="I love to cook"
        type="textarea"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="occupation"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Occupation
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="occupation"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="occupation"
        placeholder="Chef"
        type="text"
        value=""
      />
    </FormGroup>
    <Button
      className="btn-block mt-0"
      color="secondary"
      disabled={true}
      tag="button"
    >
      REGISTER
    </Button>
  </Form>
  <p
    className="text-center"
  >
    Already have an account, 
    <Link
      replace={false}
      target="_self"
      to="\\\\login"
    >
      Log in here
    </Link>
    .
  </p>
</div>
`;

exports[`SignupForm shows error alert and disables submit button when there's a submit error 1`] = `
<div
  id="register-form-div"
>
  <h4
    className="text-center"
  >
    Register for a New Account
  </h4>
  <hr />
  <p
    className="text-muted mx-auto text-center"
  >
    Fields marked
    <span
      className="text-danger"
    >
      *
    </span>
     are important
  </p>
  <Form
    className="mt-4 mb-3 px-5"
    id="register-form"
    onSubmit={[Function]}
    tag="form"
  >
    <NormalAlert
      color="danger"
    >
      <p
        className="text-center mb-0"
      >
        Error submitting form
      </p>
    </NormalAlert>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="firstname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        First Name
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="firstname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="firstname"
        placeholder="Jane"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="lastname"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Last Name
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="lastname"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="lastname"
        placeholder="Smith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="email"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Email Address
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="email"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="email"
        placeholder="janesmith@me.com"
        type="email"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="username"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Username
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="username"
        meta={
          Object {
            "asyncValidating": false,
            "error": null,
            "touched": false,
          }
        }
        name="username"
        placeholder="janesmith"
        type="text"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
       
      <small
        className="text-muted"
      >
        (more than 10 characters)
      </small>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="password"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="password"
        placeholder="••••••••••"
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        for="passwordconfirm"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Password Confirmation
        <span
          className="text-danger"
        >
          *
        </span>
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="passwordconfirm"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="passwordConfirm"
        placeholder={null}
        type="password"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="user-info"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Tell Us Something About Yourself
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="user-info"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="aboutMe"
        placeholder="I love to cook"
        type="textarea"
        value=""
      />
    </FormGroup>
    <FormGroup
      tag="div"
    >
      <Label
        className="col-form-label col-form-label-lg"
        htmlFor="occupation"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Occupation
      </Label>
      <RenderInput
        handleBlur={[Function]}
        handleChange={[Function]}
        handleFocus={[Function]}
        id="occupation"
        meta={
          Object {
            "error": null,
            "touched": false,
          }
        }
        name="occupation"
        placeholder="Chef"
        type="text"
        value=""
      />
    </FormGroup>
    <Button
      className="btn-block mt-0"
      color="secondary"
      disabled={true}
      tag="button"
    >
      REGISTER
    </Button>
  </Form>
  <p
    className="text-center"
  >
    Already have an account, 
    <Link
      replace={false}
      target="_self"
      to="\\\\login"
    >
      Log in here
    </Link>
    .
  </p>
</div>
`;
